### PR TITLE
add cmek option for cloud_run_v2

### PR DESCRIPTION
**Pull request by SalehElnagarSecurrency**
---
**Checklist**

I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

**Review by juliocc — APPROVED** [review]

File: modules/cloud-run-v2/service.tf [hunk 1]
```diff
@@ -15,14 +15,15 @@
...
-  launch_stage = var.launch_stage
+  count          = var.create_job ? 0 : 1
+  provider       = google-beta
+  project        = var.project_id
+  location       = var.region
+  name           = "${local.prefix}${var.name}"
+  ingress        = var.ingress
+  labels         = var.labels
+  launch_stage   = var.launch_stage
+  encryption_key = var.encryption_key
```
Comment (wiktorn) [inline modules/cloud-run-v2/service.tf:26]: For symmetry, can we add this right away to job, as both resources support CMEK?

**Comment by wiktorn** [comment]
@SalehElnagarSecurrency you can use `tools/tfdoc.py` in the repo to regenerate the README.md

File: modules/cloud-run-v2/service.tf (hunk 1 above)
Comment (SalehElnagarSecurrency) [reply modules/cloud-run-v2/service.tf:26]: Yes, we can.

**Comment by SalehElnagarSecurrency** [comment]
> @SalehElnagarSecurrency you can use `tools/tfdoc.py` in the repo to regenerate the README.md

I haven't cloned the repo on my local )). but you're right. It's better to do it this way.

**Comment by wiktorn** [comment]
> @SalehElnagarSecurrency can fix the failing tests and linting issues?

@SalehElnagarSecurrency my guess is that the tests are failing, as the example you have added is missing: `# tftest modules=n resources=nn` line at the end. Substitute `n` and `nn` with actual numbers of modules / resources created.

**Comment by SalehElnagarSecurrency** [comment]
> > @SalehElnagarSecurrency can fix the failing tests and linting issues?
> 
> @SalehElnagarSecurrency my guess is that the tests are failing, as the example you have added is missing: `# tftest modules=n resources=nn` line at the end. Substitute `n` and `nn` with actual numbers of modules / resources created.

what is wrong with the new example:
`[gw3] [ 29%] FAILED tests/examples/test_plan.py::test_example[modules/cloud-run-v2:Using Customer-Managed Encryption Key:1] `

I used the same example and I managed to have a successful plan.

File: modules/cloud-run-v2/service.tf (hunk 1 above)
Comment (wiktorn) [reply modules/cloud-run-v2/service.tf]: I don't see the file changed, did you forgot to commit that change?

File: modules/cloud-run-v2/README.md [hunk 2]
```diff
@@ -185,6 +186,26 @@ module "cloud_run" {
...
+
+Deploy a Cloud Run service with environment variables encrypted using a Customer-Managed Encryption Key. Ensure you specify the encryption_key with the full resource identifier of your Cloud KMS CryptoKey. This setup adds an extra layer of security by utilizing your own encryption keys.
+
+```hcl
+module "cloud_run" {
+  source         = "./fabric/modules/cloud-run-v2"
+  project_id     = var.project_id
+  region         = var.region
+  name           = "hello"
+  encryption_key = "projects/my-project/locations/location/keyRings/my-keyring/cryptoKeys/my-cryptokey"
```
Comment (wiktorn) [inline modules/cloud-run-v2/README.md:199]:
```suggestion
  encryption_key = var.kms_key.id
```
This way this example will be compatibile with E2E tests

**Comment by wiktorn** [comment]
Thanks a lot @SalehElnagarSecurrency !!

**Pull request merged**